packages/graphql/src/schema/validation/schema-validation.ts

```diff
@@ -61,30 +61,18 @@ export function validateUserDefinition(
     augmentedDocument: DocumentNode,
     additionalDirectives: Array<GraphQLDirective> = [],
     additionalTypes: Array<GraphQLNamedType> = [],
-    rules: readonly SDLValidationRule[] = specifiedSDLRules
+    rules?: readonly SDLValidationRule[]
 ): void {
+    rules = rules ? rules : [...specifiedSDLRules, DirectiveArgumentOfCorrectType];
     const validationDocument = makeValidationDocument(userDocument, augmentedDocument);
+
     const schemaToExtend = new GraphQLSchema({
         directives: [...specifiedDirectives, ...additionalDirectives],
         types: [...additionalTypes],
     });
 
-    const errors = validateSDL(validationDocument, rules, schemaToExtend);
+    const errors = validateSDL(validationDocument, [...rules, DirectiveArgumentOfCorrectType], schemaToExtend);
```

Comment:
`DirectiveArgumentOfCorrectType` is added to the rules list twice: on line 66 and here. Could maybe keep the default value `specifiedSDLRules` when defining the parameter and expand it here with the additional ones

Reply:
Good catch!